packages/framework/ai-collab/src/explicit-strategy/agentEditReducer.ts

```diff
@@ -552,6 +596,216 @@ function assertObjectIdsExist(treeEdit: TreeEdit, idGenerator: IdGenerator): voi
 	}
 }
 
+const createNodePathRecursive = (
+	node: TreeNode | undefined,
+	idGenerator: IdGenerator,
+	currentPath: NodePath,
+): NodePath => {
+	if (node === undefined) {
+		return currentPath;
+	}
+
+	currentPath.push({
+		shortId: Tree.shortId(node),
+		schemaIdentifier: Tree.schema(node).identifier,
+		parentField: Tree.key(node),
+	});
+
+	const parentNode = Tree.parent(node);
+	return createNodePathRecursive(parentNode, idGenerator, currentPath);
+};
+
+/**
+ * Creates a UI diff for an Insert TreeEdit. This should be executed AFTER an insertion is made.
```

Comment:
```suggestion
 * Creates a diff for an Insert TreeEdit. This should be executed AFTER an insertion is made.
```

Reply:
It would also be good to document _why_ this needs to be called after the insertion is applied.